db/migrate/20161110113139_tidy_up_removed_documents.rb

```diff
@@ -0,0 +1,99 @@
+# The history
+# ===========
+#
+# Following on from the previous migration where we deleted ids from the
+# removed_document_ids field, now we need to make sure that the documents
+# referred to by the remaining entries have been correctly withdrawn as if
+# the new removal functionality had been in place.
+#
+# This means setting the state to archived and discarding drafts from the
+# publishing-api for any documents in 'draft' state, and creating a new
+# edition that we archive and send an unpublishing to the publishing-api.
+require 'rummager_indexer'
+require 'formatters/manual_section_indexable_formatter'
+
+class TidyUpRemovedDocuments < Mongoid::Migration
+  def self.up
+    ManualRecord.all.to_a.each do |manual_record|
+      puts %{Collecting removed_document_editons from "#{manual_record.slug}":#{manual_record.manual_id}}
+      removed_document_ids = manual_record.editions.map do |manual_edition|
+        manual_edition.removed_document_ids || []
+      end.flatten.uniq
+      puts %{Found #{removed_document_ids.count} removed documents}
+      removed_document_ids.map do |document_id|
```

Comment:
I'm nitpicking but I don't think we need `map` here as we're not using the returned array. We could use `each`?